src/core/data-cite.js

```diff
@@ -2,29 +2,28 @@
 /**
  * Module core/data-cite
  *
- * Allows citing other specifications using
- * anchor elements. Simply add "data-cite"
- * and key of specification.
+ * Allows citing other specifications using anchor elements. Simply add
+ * "data-cite" and key of specification.
  *
- * This module simply adds the found key
- * to either conf.normativeReferences
- * or to conf.informativeReferences depending on
- * if it starts with a ! or not.
- *
- * Usage:
- * https://github.com/w3c/respec/wiki/data--cite
+ * This module links elements that have `data-cite` attributes by converting
+ * `data-cite` to `href` attributes. `data-cite` attributes are added to markup
+ * directly by the author as well as via other modules like core/xref.
  */
 import { biblio, resolveRef, updateFromNetwork } from "./biblio.js";
+import { pub, sub } from "./pubsubhub.js";
 import {
   refTypeFromContext,
   showInlineError,
   showInlineWarning,
   wrapInner,
 } from "./utils.js";
-import { sub } from "./pubsubhub.js";
 export const name = "core/data-cite";
 
-const THIS_SPEC = "__SPEC__";
+/**
+ * An arbitrary constant value used as an alias to current spec's shortname. It
+ * exists to simplify code as passing `conf.shortName` everywhere gets clumsy.
```

Comment:
How about migrating to a class as #2752 did?

Reply:
It's not just this module that makes use of `toCiteDetails`, which is why we introduced `THIS_SPEC` in https://github.com/w3c/respec/pull/2826
Migrating to class like in #2752 is helpful when accessing `conf` within same module.

Reply:
>Migrating to class like in #2752 is helpful when accessing conf within same module.

Well no, it's helpful when needing to pass `conf` again and again and that `conf` can't be kept in the same module. Accessing `conf` within same module should be replaced by class property, not `conf`.

Reply:
Sorry, not following. `THIS_SPEC`  (or earlier `conf.shortName`) is being used in `core/data-cite@toCiteDetails` as well as `core/link-to-dfn@updateReferences`. How would we access this common constant with the class approach?

Reply:
Each instance can keep `conf` as a class property value so that its methods can access it without passing it everywhere just to access `conf.shortName`.

Reply:
How do we access `toCiteDetails` outside `core/data-cite` when we've moved it into a Plugin class?

Reply:
In that case we can keep a module function version of `toCiteDetails` and then a new wrapper method with the same name to pass `conf` to the function.

That said, I didn't see this is just an existing code. No need to change it in this PR IMO.

Reply:
Well not totally, it's introduced yesterday? 😮

Reply:
Yes. Should've asked your feedback there